File: core_py/examples/add_book.py
from report_core import XLSXBook
import time


if __name__ == "__main__":
    start_time = time.time()

    book = XLSXBook()
    sheet = book.add_sheet("A", 5, 5)
    print("Sheet", sheet)

    sheet.write_cell(1, 1, "1000")

    book.get_sheet_name("A")
    print("Find Name Sheet", sheet)

    book.get_sheet_index(0)
    print("Find Idx Sheet", sheet)

    print(f"Выполнено за: {time.time() - start_time:.3f} сек.")

File: core_rs/examples/add_book.rs
use std::time::Instant;

use core_rs::writer::book::XLSXBook;

fn main() -> anyhow::Result<()> {
    let start = Instant::now();

    let book = XLSXBook::new();
    let _ = book.lock().add_sheet("A".to_string(), None, None);

    let sheet = book.lock().get_sheet_name("A".to_string());
    println!("Find NameSheet {:?}", sheet.map(|s| s.lock().name.clone()));

    let sheet = book.lock().get_sheet_index(0);
    println!("Find Idx Sheet {:?}", sheet.map(|s| s.lock().name.clone()));

    let end = start.elapsed();
    println!(
        "Выполнено за: {}.{:03} сек.",
        end.as_secs(),
        end.subsec_millis(),
    );

    Ok(())
}
